### ISSUE TITLE
DllNotFoundException on Android

### ISSUE THREAD

**Issue by TonSharp** [gilzoide/unity-lottie-player#8]
Hi! I am using Firebase Crashlytics to track crashes and non-fatals and I am observing the following issue. On many Android devices (especially on Samsung devices), the following exception occurs: 

```
Non-fatal Exception: java.lang.Exception: DllNotFoundException: Unable to load DLL 'lottie-player'. Tried to load the following dynamic libraries: Unable to load dynamic library 'lottie-player' because of 'Failed to open the requested dynamic library (0x06000000) dlerror() = dlopen failed: library "lottie-player" not found
at Gilzoide.LottiePlayer.RLottie.RLottieCApi.lottie_init(Gilzoide.LottiePlayer.RLottie.RLottieCApi)
```

With this stacktrace:

```
at Gilzoide.LottiePlayer.RLottie.RLottieCApi.lottie_init(Gilzoide.LottiePlayer.RLottie.RLottieCApi)
       at Gilzoide.LottiePlayer.RLottie.RLottieCApi..cctor(Gilzoide.LottiePlayer.RLottie.RLottieCApi.)
       at Gilzoide.LottiePlayer.NativeLottieAnimation..ctor(Gilzoide.LottiePlayer.NativeLottieAnimation.)
       at Gilzoide.LottiePlayer.LottieAnimationAsset.CreateNativeAnimation(Gilzoide.LottiePlayer.LottieAnimationAsset)
       at Gilzoide.LottiePlayer.ImageLottiePlayer.RecreateAnimationIfNeeded(Gilzoide.LottiePlayer.ImageLottiePlayer)
       at Gilzoide.LottiePlayer.ImageLottiePlayer.OnEnable(Gilzoide.LottiePlayer.ImageLottiePlayer)
```

What could be causing the issue with loading the library? And could the following link.xml file with the specified lines help in this case:

```xml
<linker>
    <assembly fullname="lottie-player" preserve="all"/>
</linker>
```

**Comment by TonSharp** [comment]
Also I've noticed that all the crashes are coming from the ARMv7 VFPv3 NEON processor. Maybe this will help to solve the problem

**Comment by gilzoide** [comment]
Hey @TonSharp, thanks for the report.
Hmm, weird.

> Also I've noticed that all the crashes are coming from the ARMv7 VFPv3 NEON processor. Maybe this will help to solve the problem

Well, the rlottie library does use NEON instructions, and it has even some assembly code for ARMv7 (specifically, the [pixman](https://github.com/Samsung/rlottie/tree/master/src/vector/pixman) code). I haven't tried this library on any ARMv7 devices, only ARM64, likely the build is wrong for this architecture, sorry about that 😔 I don't think I added this assembly to my amalgamated C++ code.
I do have a spare Android device with an ARMv7 CPU, so I can try in it later.

**Comment by gilzoide** [comment]
> This is not relates to the issue topic but also needs to be discussed

Then please create a new issue for it, so that we can discuss it properly.

**Comment by TonSharp** [comment]
Hi @gilzoide , is there any news about non fatals on android?

**Comment by gilzoide** [comment]
No, I haven't tryed it yet. It's quite possible that fixing the build for armv7 including the assembly code will fix it, I'll try looking into it this week.

**Comment by TonSharp** [comment]
Okay, thank you. I hope the fix will help and I can integrate the new version in the next release of my app. Good luck!

**Comment by gilzoide** [comment]
Hey @TonSharp, I've confirmed that the problem was building without including the pixman arm neon assembly code.
I had problems building it with the Android NDK 26, so instead I'm using a custom version of rlottie that just ignores the custom neon code when building with clang. I've tested the fix in a real Android device with arm32 CPU, and it works now!
I'll relase this fix as 1.0.1 soon, so you can update in your app.
Please let me know if you still have problems after this.

Cheers \o/
